### PR TITLE
Bring boilerplate code up to date

### PR DESCRIPTION
**Pull request by codegard1**
This covers a range of changes I made to "modernize" the configuration. I had to make some of these changes to overcome issues I had while trying to debug the extension locally. I also updated dependencies and package catagories in `package.json`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
     "compile": "node ./node_modules/vscode/bin/compile -watch -p ./"
   },
   "devDependencies": {
-    "typescript": "^3.1.6",
+    "typescript": "^3.2.1",
```
Comment (cgay) [inline package.json]: This seems like a very specific version to depend on. Is there some particular feature we need in that version?  (I know nothing about the VS Code / Typescript ecosystem, so I'm just asking...

File: .vscode/settings.json [hunk 2]
```diff
@@ -7,5 +7,6 @@
     "out": true // set this to false to include "out" folder in search results
   },
   "typescript.tsdk": "./node_modules/typescript/lib",
-  "restructuredtext.confPath": "" // we want to use the TS server from our node_modules folder to control its version
-}
+  "restructuredtext.confPath": "", // we want to use the TS server from our node_modules folder to control its version
+  "typescript.tsc.autoDetect": "off" // Turn off tsc task auto detection since we have the necessary task as npm scripts
+}
```
[the remark this answers is not in the document]
Comment (pedro-w) [reply .vscode/settings.json]: Standard JSON doesn't have '//' comments like that!

[the remark this answers is not in the document]
Comment (pedro-w) [reply .vscode/settings.json]: "This file is intended to be committed into source repositories" [(source)](https://docs.npmjs.com/files/package-lock.json)

[the remark this answers is not in the document]
Comment (cgay) [reply .vscode/settings.json]: > Standard JSON doesn't have '//' comments like that!

hah!  And I wrote most of our json parser.

(Ours has a "strict?" mode in which comments aren't allowed and non-strict in which '#' is the comment character.)

[the remark this answers is not in the document]
Comment (codegard1) [reply .vscode/settings.json]: @cgay `package-lock.json` is a detailed map of exactly which version of every single dependency in the dep tree is being used for this version of the package. It is meant to help ensure the environment is always the same when you run it.

File: .vscode/launch.json [hunk 3]
```diff
@@ -7,22 +6,31 @@
...
+        "--extensionDevelopmentPath=${workspaceFolder}",
+        "--extensionTestsPath=${workspaceFolder}/out/test"
+      ],
       "stopOnEntry": false,
       "sourceMaps": true,
-      "outFiles": ["out/test/dylanMain.test.ts"],
+      "outFiles": [
+        "out/test/dylanMain.test.ts"
+      ],
       "preLaunchTask": "npm"
```
[the remark this answers is not in the document]
Comment (codegard1) [reply .vscode/launch.json]: Yes, it should.

**Comment by codegard1** [comment]
@cgay `package-lock.json` is meant to track the version of every installed dependency in the dependency tree. It's supposed to help with consistency. 

See also: https://docs.npmjs.com/files/package-lock.json

File: package.json (hunk 1 above)
Comment (codegard1) [reply package.json]: No, there is not a good rationale for this. I will attempt to find one.

Comment (codegard1) [reply package.json]: @housel has the solution [here](https://gist.github.com/pedro-w/9558d1d5de14ef7dda247c588fd088f8).

The current minimum version of TS supported by the extension now is 2.0.3. So I will update `package.json` accordingly.

**Comment by cgay** [comment]
Chris is going to do a new PR. Closing.

**Pull request closed without merging**